File: solutions/485.max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int res = 0;
        int cnt = 0;
        for(int num : nums){
            if(num == 0){
                if(cnt != 0){
                    res = Math.max(res, cnt);
                    cnt = 0;
                }
                 
            }else{
                cnt++;
            }
        }
        res = Math.max(res, cnt);
        return res;
    }
}

File: solutions/485.max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let res = 0;
    let cnt = 0;
    for(let num of nums){
        if(num === 1){
            cnt++;
        }else if(cnt != 0){
            res = Math.max(res, cnt);
            cnt = 0;
        }
    }
    res = Math.max(res, cnt);
    return res;
};